server/server_listen.go

```diff
@@ -82,13 +85,41 @@ func (s *Server) tlsLetsEncrypt(domains []string) *tls.Config {
 	return m.TLSConfig()
 }
 
-func tlsKeyCert(key, cert string) (*tls.Config, error) {
+func (s *Server) tlsKeyCert(key, cert string, clientCaPath string) (*tls.Config, error) {
 	c, err := tls.LoadX509KeyPair(cert, key)
 	if err != nil {
 		return nil, err
 	}
-	//return file based tls config using tls defaults
-	return &tls.Config{
-		Certificates: []tls.Certificate{c},
-	}, nil
+	//mTLS requires Client CAs
+	if clientCaPath != "" {
+		files, err := ioutil.ReadDir(clientCaPath)
```

Comment:
CAs can be encoded into bundles. So you can have multiple CAs concatenated in a single file.

Reply:
I thought about this when I work on this. In reality, most of the users will put individual client certs (self signed) over here instead of one trusted CA. If we only allow one CA bundle, that means users have to maintain a combined CA bundle by themselves, of which mostly likely they don't know how to do it. Even they can handle it, adding and removing certs will be a hassle. Please give a thought on this.

Reply:
Okay, fair enough. Though it should also support single files. Can `os.Stat` and then `if IsDir()` do your logic, `else` load single file.

Reply:
sounds good